Client/Frontend/Browser/Search/BraveSearchManager.swift

```diff
@@ -0,0 +1,293 @@
+                let canAnswerResponse = try JSONDecoder().decode(BackupQuery.self, from: output.data)
+                
+                return canAnswerResponse
+            }
+            .sink(receiveCompletion: { status in
+                switch status {
+                case .failure(let error):
+                    log.error("shouldUseFallback error: \(error)")
+                    completion(nil)
+                case .finished:
+                    // Completion is called on `receiveValue`.
+                    break
+                }
+            }, receiveValue: { canAnswer in
+                completion(canAnswer)
+            })
+            .store(in: &cancellables)
+    }
+    
+    /// Perform a backup search using an alternative search engine, gets results back as html source.
+    func backupSearch(with backupQuery: BackupQuery,
+                      completion: @escaping (String) -> Void) {
+        
+        guard var components = URLComponents(string: fallbackProviderURLString) else { return }
+        
+        var queryItems: [URLQueryItem] = [
+            .init(name: "q", value: query.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed))]
+        
+        if let language = backupQuery.language {
+            queryItems.append(.init(name: "hl", value: language))
+        }
+        
+        if let country = backupQuery.country {
+            queryItems.append(.init(name: "gl", value: country))
+        }
+        
+        components.queryItems = queryItems
+        
+        guard let url = components.url else { return }
+        var request = URLRequest(url: url, timeoutInterval: 3)
+        
+        // Must be set, without it the fallback results may be not retrieved correctly.
+        request.addValue(UserAgent.userAgentForDesktopMode, forHTTPHeaderField: "User-Agent")
+        
+        fallbackQueryResultsPending = true
+        URLSession(configuration: .ephemeral)
+            .dataTaskPublisher(for: request)
+            .tryMap { output -> String in
+                guard let response = output.response as? HTTPURLResponse,
+                      let contentType = response.value(forHTTPHeaderField: "Content-Type"),
+                      response.statusCode >= 200 && response.statusCode < 300 else {
+                    throw "Invalid response"
+                }
+                
+                // For some reason sometimes no matter what headers are set, ISO encoding is returned
+                // we check for iso and fallback to utf8 by default.
+                let encoding: String.Encoding =
+                    contentType.contains("ISO-8859-1") ? .isoLatin1 : .utf8
+                
+                guard let stringFromData = String(data: output.data, encoding: encoding) else {
+                    throw "Failed to decode string from data"
+                }
+                
+                return stringFromData.javaScriptEscapedString
+            }
+            .eraseToAnyPublisher()
```

Comment:
Don't think this is needed if you are not returning the publisher anywhere as `AnyPublisher`